### DISCUSSION TITLE
执行build.sh报错

### DISCUSSION THREAD

**Discussion by Suzzt** [apache/doris#27300] [A - General / Q&A]
执行build.sh 报错（最新代码，master分支，本地/Library/Java/JavaVirtualMachines/jdk1.8.0_211.jdk/Contents/Home/include/jni.h有这个文件啊）

======错误======
FAILED: CMakeFiles/pch.dir/cmake_pch.hxx.pch 
ccache /usr/local/opt/llvm@16/bin/clang++ -DBOOST_STACKTRACE_USE_NOOP -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_LIBCPP -DUSE_LIBHDFS3 -I/Users/sucf/IdeaProjects/doris/be/src/apache-orc/c++/include -I/Users/sucf/IdeaProjects/doris/be/build_Release/src/apache-orc/c++/include -I/Users/sucf/IdeaProjects/doris/be/build_Release/src/clucene/src/shared -I/Users/sucf/IdeaProjects/doris/be/src/clucene/src/core -I/Users/sucf/IdeaProjects/doris/be/src/clucene/src/shared -I/Users/sucf/IdeaProjects/doris/be/src/clucene/src/contribs-lib -I/Users/sucf/IdeaProjects/doris/be/src -I/Users/sucf/IdeaProjects/doris/be/test -I/include -I/include/darwin -isystem /Users/sucf/IdeaProjects/doris/be/../gensrc/build -isystem /Users/sucf/IdeaProjects/doris/thirdparty/installed/include -isystem /Users/sucf/IdeaProjects/doris/thirdparty/installed/gperftools/include -O3 -DNDEBUG  -O3 -DNDEBUG -std=gnu++20 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk -mmacosx-version-min=12.6   -D OS_MACOSX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fcolor-diagnostics -Wpedantic -Wshadow-field -Wunused -Wunused-command-line-argument -Wunused-exception-parameter -Wunused-volatile-lvalue -Wunused-template -Wunused-member-function -Wunused-macros -Wno-vla-extension -Wno-gnu-statement-expression -stdlib=libc++ -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -mavx2 -DUSE_JEMALLOC -DENABLE_STACKTRACE -Xclang -fno-pch-timestamp -Winvalid-pch -fpch-instantiate-templates -Xclang -emit-pch -Xclang -include -Xclang /Users/sucf/IdeaProjects/doris/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx -x c++-header -MD -MT CMakeFiles/pch.dir/cmake_pch.hxx.pch -MF CMakeFiles/pch.dir/cmake_pch.hxx.pch.d -o CMakeFiles/pch.dir/cmake_pch.hxx.pch -c /Users/sucf/IdeaProjects/doris/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx.cxx
In file included from <built-in>:1:
In file included from /Users/sucf/IdeaProjects/doris/be/build_Release/CMakeFiles/pch.dir/cmake_pch.hxx:5:
/Users/sucf/IdeaProjects/doris/be/src/pch/pch.h:471:10: fatal error: 'jni.h' file not found
#include <jni.h>
         ^~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
======错误======

**Comment by LemonLiTree** [comment]
Confirm whether you are using the latest third-party library
https://github.com/apache/doris-thirdparty/releases/tag/automation

**Reply by Suzzt** [reply]
我昨天排查到了。是文件权限问题，不知道为什么我doris工程下有些文件是root权限，真是。。。
现在build确实执行下去了，很慢，但是又报了一个错误，不过感觉我快成功了～
======错误======
node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
Error: npm is not found
======错误======
bwt，"node -v", "npm -v"输出的分别是v6.17.1,   3.10.10.  是因为版本问题吗？

**Reply by LemonLiTree** [reply]
yeah

**Reply by Suzzt** [reply]
但是我尝试更新到了最新的node版本。也是还是这个报错

**Reply by LemonLiTree** [reply]
https://doris.apache.org/zh-CN/docs/dev/install/source-install/compilation-with-ldb-toolchain 
try v12.13.0

**Reply by Suzzt** [reply]
我切到这个版本后。依然是这个报错，不知道哪里还有问题是我没注意到的
<img width="826" alt="image" src="https://github.com/apache/doris/assets/27397567/bf316ef6-dc91-45d2-bb9e-14e9409a4d95">

**Reply by LemonLiTree** [reply]
npm -v

**Reply by Suzzt** [reply]
<img width="485" alt="image" src="https://github.com/apache/doris/assets/27397567/2ad82b23-bd32-465b-b28f-d6912c8b6d1b">

**Reply by LemonLiTree** [reply]
a bit strange

**Reply by 521alice** [reply]
> 我昨天排查到了。是文件权限问题，不知道为什么我doris工程下有些文件是root权限，真是。。。 现在build确实执行下去了，很慢，但是又报了一个错误，不过感觉我快成功了～ ======错误====== node: --openssl-legacy-provider is not allowed in NODE_OPTIONS Error: npm is not found ======错误====== bwt，"node -v", "npm -v"输出的分别是v6.17.1, 3.10.10. 是因为版本问题吗？

请问这一步是怎么解决的，我是mac目前也是卡在这里

**Reply by xu20160924** [reply]
@Suzzt Have you solved that?
